docs/reference/processors.md

```diff
@@ -0,0 +1,196 @@
+
+### Analyzing generation results
+
+Once tracking is enabled, you can analyze the generation process in several ways:
+
+1. Get the top tokens at each position:
+
+```python
+# Get top 5 tokens at each position
+top_k = generator.logits_processor.get_top_tokens(k=5)
+
+# Analyze each position
+for position_dict in top_k:
+    print(f"\nPosition {position_dict['position']}:")
+    print(f"Text so far: {position_dict['text_so_far']}")
+
+    for token in position_dict['tokens']:
+        print(f"\nToken: {token['token']}")
+        print(f"Unstructured probability: {token['unstructured_prob']:.3f}")
+        print(f"Structured probability: {token['structured_prob']:.3f}")
+        print(f"Unstructured logit: {token['unstructured_logit']:.3f}")
+        print(f"Structured logit: {token['structured_logit']:.3f}")
+        print(f"Was chosen: {token['is_chosen']}")
+```
+
+2. Convert to a pandas DataFrame for analysis:
+
+```python
+import pandas as pd
+
+# Get all tokens with probability > 1%
+df = generator.logits_processor.to_dataframe(show="probs", min_value=0.01)
+print(df)
+#    position token   natural  constrained  chosen
+# 0         0   You  0.021324          0.0   False
+# 1         0   The  0.021959          0.0   False
+# 2         0  Sure  0.025492          0.0   False
+# 3         0  JSON  0.031045          0.0   False
+# 4         0    To  0.031047          0.0   False
+```
+
+3. Get the generated sequence up to a position:
+
+```python
+# Get text generated up to position 5
+text = generator.logits_processor.sequence(5)
+```
+
+### Memory management
+
+The tracking processor stores logits in memory for analysis, and offloads logits to main memory if you use a GPU. For long sequences, you have several options:
+
+1. Clear tracking data when no longer needed:
+```python
+generator.logits_processor.clear()
+```
+
+2. Filter data when analyzing:
+```python
+# Only analyze specific positions
+results = generator.logits_processor.get_top_tokens(positions=[0, 1, 2])
+
+# Only look at high probability tokens
+df = generator.logits_processor.to_dataframe(show="probs", min_value=0.01)
+```
+
+### Important notes about logit tracking
+
+- Tracking logits is a slow operation, so do not use it in production environments
+- The processor will accumulate logits if you call `generator(prompt)` multiple times, meaning that the tokens stored can be aggregated across generations. You can use `generator.logits_processor.clear()` to reset the processor, or construct a new generator and call `add_tracking` again to start tracking from scratch.
+- Processed logits will contain `-inf` values when structured outputs are used
+- Token decoding requires the wrapped processor to have a tokenizer attribute
+- Memory usage grows linearly with sequence length
+- The tracking processor only supports single-batch processing
+- Tracking logits can incur significant overhead -- do not use it in production environments
```

Comment:
I would add an example of how we can use these logits processors directly with e.g. transformer pipes